langchain4j-azure-cosmos/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/AzureCosmosDbMongoVCoreEmbeddingStore.java

```diff
@@ -0,0 +1,568 @@
+     *                                it manually.
+     * @param kind                    - Type of vector index to create.
+     *                                Possible options are:
+     *                                - vector-ivf
+     *                                - vector-hnsw: available as a preview feature only, to enable visit
+     *                                https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/preview-features
+     * @param numLists                - This integer is the number of clusters that the inverted file (IVF) index uses to group the
+     *                                vector data. We recommend that numLists is set to documentCount/1000 for up to 1 million
+     *                                documents and to sqrt(documentCount) for more than 1 million documents. Using a numLists value
+     *                                of 1 is akin to performing brute-force search, which has limited performance.
+     * @param similarity              - Similarity metric to use with the index.
+     *                                Possible options are:
+     *                                - COS (cosine distance),
+     *                                - L2 (Euclidean distance), and
+     *                                - IP (inner product).
+     * @param dimensions              - Number of dimensions for vector similarity. The maximum number of supported dimensions
+     *                                is 2000.
+     * @param numberOfConnections     - The max number of connections per layer (16 by default, minimum value is 2, maximum
+     *                                value is 100). Higher m is suitable for datasets with high dimensionality and/or high
+     *                                accuracy requirements.
+     * @param efConstruction          - the size of the dynamic candidate list for constructing the graph (64 by default, minimum
+     *                                value is 4, maximum value is 1000). Higher ef_construction will result in better index
+     *                                quality and higher accuracy, but it will also increase the time required to build the index.
+     *                                ef_construction has to be at least 2 * m.
+     * @param efSearch                - The size of the dynamic candidate list for search (40 by default). A higher value provides
+     *                                better recall at the cost of speed.
+     */
+    public AzureCosmosDbMongoVCoreEmbeddingStore(
+            MongoClient mongoClient,
+            String endpoint,
+            String databaseName,
+            String collectionName,
+            String indexName,
+            String applicationName,
+            CreateCollectionOptions createCollectionOptions,
+            Boolean createIndex,
+            String kind,
+            int numLists,
```

Comment:
please use object types in builder instead of primitives

Reply:
Updated the code